rotkehlchen/tests/unit/decoders/test_weth.py

```diff
@@ -465,3 +475,516 @@ def test_wxdai_wrap(database, gnosis_inquirer, gnosis_accounts):
+            sequence_index=0,
+            timestamp=timestamp,
+            location=Location.ARBITRUM_ONE,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=A_ETH,
+            balance=Balance(amount=FVal('0.000001382891214')),
+            location_label='0x7aBAee8F04EFd689961115f7A28bAA2E73Be6703',
+            notes='Burned 0.000001382891214 ETH for gas',
+            counterparty=CPT_GAS,
+        ),
+        EvmEvent(
+            tx_hash=evmhash,
+            sequence_index=1,
+            timestamp=timestamp,
+            location=Location.ARBITRUM_ONE,
+            event_type=HistoryEventType.DEPOSIT,
+            event_subtype=HistoryEventSubType.DEPOSIT_ASSET,
+            asset=A_ETH,
+            balance=Balance(amount=FVal(0.007767825959188763)),
+            location_label='0x7aBAee8F04EFd689961115f7A28bAA2E73Be6703',
+            notes='Wrap 0.007767825959188763 ETH in WETH',
+            counterparty=CPT_WETH,
+            address=string_to_evm_address('0x82aF49447D8a07e3bd95BD0d56f35241523fBab1'),
+        ),
+        EvmEvent(
+            tx_hash=evmhash,
+            sequence_index=2,
+            timestamp=timestamp,
+            location=Location.ARBITRUM_ONE,
+            event_type=HistoryEventType.RECEIVE,
+            event_subtype=HistoryEventSubType.RECEIVE_WRAPPED,
+            asset=A_WETH_ARB,
+            balance=Balance(amount=FVal(0.007767825959188763)),
+            location_label='0x0000000000000000000000000000000000000000',
+            notes='Receive 0.007767825959188763 WETH',
+            counterparty=CPT_WETH,
+            address=string_to_evm_address('0x82aF49447D8a07e3bd95BD0d56f35241523fBab1'),
+        ),
+    ]
+    assert events == expected_events
+
+
+@pytest.mark.vcr()
+@pytest.mark.parametrize('optimism_accounts', [['0x81aa5101D4c376cd6DC031EA62D7b64A9BAE10a0']])
+def test_weth_withdraw_optimism(database, optimism_inquirer):
+    """
+    Data for tx is taken from
+    https://optimistic.etherscan.io/tx/0x4a6b47e1f622a8ad059bd0723c53f2c71f12e7b105d2ef2ff4dff07ac1f185c0
+    """
+    evmhash = deserialize_evm_tx_hash(
+        '0x4a6b47e1f622a8ad059bd0723c53f2c71f12e7b105d2ef2ff4dff07ac1f185c0',
+    )
+    events, _ = get_decoded_events_of_transaction(
+        evm_inquirer=optimism_inquirer,
+        database=database,
+        tx_hash=evmhash,
+    )
+    timestamp = TimestampMS(1712240095000)
+
+    assert len(events) == 3
+    expected_events = [
+        EvmEvent(
+            tx_hash=evmhash,
+            sequence_index=0,
+            timestamp=timestamp,
+            location=Location.OPTIMISM,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=A_ETH,
+            balance=Balance(amount=FVal('0.000001897927938075')),
+            location_label='0x81aa5101D4c376cd6DC031EA62D7b64A9BAE10a0',
+            notes='Burned 0.000001897927938075 ETH for gas',
```

Comment:
To avoid repeating the values in balance and notes check what we do here https://github.com/rotki/rotki/blob/341b2ed4c1926190b40f5c748d7074192eab76c9/rotkehlchen/tests/unit/decoders/test_aave_v3.py#L39 . Do a similar thing in the tests on this file